dbt/task/deps.py

```diff
@@ -14,250 +17,325 @@
-
-            if target_version is None:
-                logger.error(
-                    'Could not find a matching version for package {}!'
-                    .format(package))
-                logger.error(
-                    '  Requested range: {}'.format(version_range))
-                logger.error(
-                    '  Available versions: {}'.format(
-                        ', '.join(available_versions)))
-                raise Exception('bad')
-
-            visited_listing.incorporate(
-                package,
-                [VersionSpecifier.from_version_string(target_version)])
-
-            target_version_metadata = dbt.clients.registry.package_version(
-                package, target_version)
 
-            dependencies = target_version_metadata.get('dependencies', {})
-
-            for package, versions in dependencies.items():
-                listing.incorporate(
-                    package,
-                    [VersionSpecifier.from_version_string(version)
-                     for version in versions])
-
-        for package, version_specifiers in visited_listing.items():
-            version_string = version_specifiers[0].to_version_string(True)
-            version_info = dbt.clients.registry.package_version(
-                package, version_string)
-
-            import requests
-
-            tar_path = os.path.realpath('{}/downloads/{}.{}.tar.gz'.format(
-                self.project['modules-path'],
-                package,
-                version_string))
-
-            logger.info("Pulling {}@{} from hub.getdbt.com...".format(
-                package, version_string))
-
-            dbt.clients.system.make_directory(
-                os.path.dirname(tar_path))
-
-            response = requests.get(version_info.get('downloads').get('tarball'))
-
-            with open(tar_path, 'wb') as handle:
-                for block in response.iter_content(1024*64):
-                    handle.write(block)
-
-            import tarfile
-
-            with tarfile.open(tar_path, 'r') as tarball:
-                tarball.extractall(self.project['modules-path'])
+class DepsTask(BaseTask):
+    def _check_for_duplicate_project_names(self, final_deps):
+        seen = set()
+        for _, package in final_deps.items():
+            project_name = package.get_project_name(self.project)
+            if project_name in seen:
+                dbt.exceptions.raise_dependency_error(
+                    'Found duplicate project {}. This occurs when a dependency'
+                    ' has the same project name as some other dependency.'
+                    .format(project_name))
+            seen.add(project_name)
 
-            logger.info(" -> Success.")
+    def run(self):
+        dbt.clients.system.make_directory(self.project['modules-path'])
+        dbt.clients.system.make_directory(DOWNLOADS_PATH)
+
+        packages = _read_packages(self.project)
+        if not packages:
+            return
```

Comment:
Can we log something like:
```
Warning: No packages found in dbt_project.yml
```

Reply:
Sure - so long as you want that. Are you figure the most common case will include at least one package?

Reply:
Oh my comment was stupid. Nevermind.